third_party/move/tools/move-asm/src/compiler.rs

```diff
@@ -0,0 +1,769 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implements the compiler from the Move assembler syntax into bytecode,
```

Comment:
nit: I was expecting this to be called an assembler.

Reply:
Good point, renamed the type but not yet the module to avoid rebase conflicts.